### PR TITLE
Upgrade scrypt-ts to v0.1.5-beta.4

### PR DESCRIPTION
**Pull request by gitzhou**

### CODE REVIEW

**Comment by gitzhou** [comment]
Fix #9

File: src/contracts/acs.ts [hunk 1]
```diff
@@ -18,23 +15,21 @@ export class AnyoneCanSpend extends SmartContract {
     @method()
     public unlock(txPreimage: SigHashPreimage, outputAmount: bigint) {
 
-        assert(this.checkPreimageSigHashType(txPreimage, SigHash.ANYONECANPAY_SINGLE));
-
-        let lockingScript: string = Utils.buildPublicKeyHashScript(this.pubKeyHash);
-        let output: string = Utils.buildOutput(lockingScript, outputAmount);
-        assert(hash256(output) == SigHash.hashOutputs(txPreimage));
+        assert(this.checkPreimageSigHashType(txPreimage, SigHash.ANYONECANPAY_SINGLE), 'preimage check failed');
```
Comment (xhliu) [inline src/contracts/acs.ts]: Use this syntax and remove `checkPreimage*` everywhere. May do it in another diff https://github.com/sCrypt-Inc/scryptTS-examples/issues/11

https://github.com/sCrypt-Inc/scrypt-ts/blob/4c45b5ded1a8ae625e8d57ebad6211bc570b0d4c/test/contracts/acs.ts#L17

File: src/contracts/advancedCounter.ts [hunk 2]
```diff
@@ -24,31 +21,27 @@ export class AdvancedCounter extends SmartContract {
         // - amount is same as specified
         // - output script is the same as scriptCode except the counter was incremented
         let amount: bigint = SigHash.value(txPreimage);
-        assert(this.updateStateSigHashType(txPreimage, amount, SigHash.ANYONECANPAY_SINGLE));
-
+        assert(this.updateStateSigHashType(txPreimage, amount, SigHash.ANYONECANPAY_SINGLE), 'stateSigHashType update failed');
```
Comment (xhliu) [inline src/contracts/advancedCounter.ts]: Same here.

**Review by xhliu — CHANGES_REQUESTED** [review]
Almost there.

File: src/contracts/ackermann.ts [hunk 3]
```diff
@@ -1,18 +1,15 @@
-import { method, prop, SmartContract, assert, int2str, len, unpack, bsv } from "scrypt-ts";
-import { UTXO } from "../types";
+import {assert, bsv, ByteString, int2str, len, method, prop, SmartContract, unpack} from "scrypt-ts";
+import {UTXO} from "../types";
 
 
 export class Ackermann extends SmartContract {
 
+    static readonly LoopCount: number = 14;
```
Comment (zhfnjust) [inline src/contracts/ackermann.ts]: use `LOOP_COUNT`

**Review by zhfnjust — APPROVED** [review]
good work

**Pull request merged**